lib/costellation.dart

```diff
@@ -0,0 +1,642 @@
+        _pointer,
+        directory.toNativeUtf8().cast<Char>(),
+        item.toNativeUtf8().cast<Char>());
+
+    if (result.error != nullptr) {
+      throw WarpException(result.error);
+    }
+    bindings.item_free(result.data);
+  }
+
+  Pointer<G_Directory> pointer() {
+    return _pointer;
+  }
+
+  void drop() {
+    bindings.directory_free(_pointer);
+  }
+}
+
+class Constellation {
+  late Pointer<G_ConstellationAdapter> _pointer;
+  Constellation(this._pointer);
+
+  Directory newDirectory(String name) {
+    Pointer<G_Directory> pointer =
+        bindings.directory_new(name.toNativeUtf8().cast<Char>());
+
+    return Directory(pointer);
+  }
+
+  File newFile(String name) {
+    Pointer<G_File> pointer =
+        bindings.file_new(name.toNativeUtf8().cast<Char>());
+
+    return File(pointer);
+  }
+
+  void createDirectoryInFilesystem(String path) {
+    G_FFIResult_Null result = bindings.constellation_create_directory(
+        _pointer, path.toNativeUtf8().cast<Char>(), false);
+    if (result.error != nullptr) {
+      throw WarpException(result.error);
+    }
+    bindings.free(result.data);
+  }
+
+  Constellation.createRecursiveDirectoryInFilesystem(String path) {
+    G_FFIResult_Null result = bindings.constellation_create_directory(
+        _pointer, path.toNativeUtf8().cast<Char>(), true);
+
+    if (result.error != nullptr) {
+      throw WarpException(result.error);
+    }
+    bindings.free(result.data);
+  }
+
+  Directory getCurrentDirectory() {
+    Pointer<G_Directory> pointerDirectory =
+        bindings.constellation_current_directory(_pointer);
+
+    if (pointerDirectory == nullptr) {
+      throw Exception("Directory not found");
+    }
+    Directory dir = Directory(pointerDirectory);
+    return dir;
+  }
+
+  Directory getCurrentDirectoryMutable() {
+    Pointer<G_Directory> pointerDirectory =
+        bindings.constellation_current_directory_mut(_pointer);
+
+    if (pointerDirectory == nullptr) {
+      throw Exception("Directory not found");
+    }
+    bindings.directory_free(pointerDirectory);
+    return Directory(pointerDirectory);
+  }
+
+  String exportConstellationInOtherTypes(ConstellationDataType type) {
+    G_FFIResult_String result =
+        bindings.constellation_export(_pointer, type.index);
+
+    if (result.error != nullptr) {
+      throw WarpException(result.error);
+    }
+
+    String data = result.data.cast<Utf8>().toDartString();
+
+    calloc.free(result.data);
+    return data;
+  }
+
+  void downloadFileFromFilesystem(String remotePath, String localPath) {
+    G_FFIResult_Null result = bindings.constellation_get(
+        _pointer,
+        remotePath.toNativeUtf8().cast<Char>(),
+        localPath.toNativeUtf8().cast<Char>());
+
+    if (result.error != nullptr) {
+      throw WarpException(result.error);
+    }
+    bindings.free(result.data);
+  }
+
+  List<int> downloadFileFromFilesystemIntoBuffer(String remotePath) {
+    List<int> buffer = [];
```

Comment:
I would have to double check but wouldnt this be a `Uint8` or maybe we could use `Uint8List`?